### DISCUSSION TITLE
Serving exported website from firebase static hosting

### DISCUSSION THREAD

**Discussion by Shrivallabh** [vercel/next.js#21562] [Help] [answered]
When serving [example](https://github.com/vercel/next.js/tree/canary/examples/with-static-export) using `serve out` the static serving works fine.

However, when uploaded to firebase hosting, the deep links do not work.

For example, when clicking on http://localhost:5000/post/1 from the index page, the post gets loaded correctly.

However, if we directly try to hit a link to http://localhost:5000/post/1 on main page, we get a 404. 

Directly hitting http://localhost:5000/post/1.html works. 

We do not want one method working in dev and one method working in final deployment.

What is the correct way to configure the statically exported projects to work on hosting such as firebase? Or is it meant to work only on Vercel platform?

I did various experiments with [exportPathMap](https://nextjs.org/docs/api-reference/next.config.js/exportPathMap) but none seemed to work so far.

**Comment by Shrivallabh** [comment, marked as answer]
I figured the following hack. I am using redirects in firebase by adding following to firebase.json

```
"redirects": [ {
            "regex": "/post/(?P<slug>[A-Za-z]*)$",
            "destination": "/post/:slug.html",
            "type": 301
        }]
```

This seems like a stop gap solution. We need a better one. Is there any option available?

**Reply by michrome** [reply]
I haven't tried it but Firebase's [`cleanUrls`](https://firebase.google.com/docs/hosting/full-config#control_html_extensions) might help.

**Reply by Shrivallabh** [reply]
@michrome Could not mark your reply as the answer. When I had read the documentation, I thought the functionality would work exactly the opposite of what I wanted.

Thanks for the answer. It is exactly what the doc ordered!
